airflow/jobs.py

```diff
@@ -1004,6 +1004,11 @@ def _execute_task_instances(self,
             .all()
         )
 
+        for task_instance in task_instances_to_examine:
```

Comment:
I'm afraid that placing this code in the main loop of the scheduler can have a serious impact on performance.

Reply:
What do you think about this place? This will not change the number of queries because the status must still be updated
https://github.com/lyft/incubator-airflow/blob/a6b08261c69a99f8253fa5f90539e8f8dadb817f/airflow/jobs.py#L1664

Reply:
But this is still not the best place in my opinion. This is only good.

Reply:
https://github.com/lyft/incubator-airflow/blob/a6b08261c69a99f8253fa5f90539e8f8dadb817f/airflow/models.py#L4520
It looks better.

Reply:
Thanks for taking a look.

I like https://github.com/lyft/incubator-airflow/blob/a6b08261c69a99f8253fa5f90539e8f8dadb817f/airflow/jobs.py#L1664  better since we are applying the policy right when we scheduled the task.

Reply:
verify_integrity is in my opinion a better place, because this is where the TaskInstance object is created. Thanks to this, we will never have incorrect information in the database.  

If we use the first solution, we have some limitations.. If we use task_policy to set owners, they will not have the correct value until it is scheduled.

Reply:
👍  That make sense.